### PR TITLE
Keras 2.1.3 port follow up

### PR DESCRIPTION
**Pull request by javierluraschi**
A few follow-ups on the Keras 2.1.3 port, few missing params and doc changes were missed.

### CODE REVIEW

File: R/model.R [hunk 1]
```diff
@@ -525,28 +524,41 @@ predict.keras.engine.training.Model <- function(object, x, batch_size=NULL, verb
...
-  object$predict_proba(
+predict_proba <- function(object, x, batch_size = NULL, verbose = 0, steps = NULL) {
+  args <- list(
     x = keras_array(x),
     batch_size = as.integer(batch_size),
     verbose = as.integer(verbose)
   )
+  
+  if (keras_version() >= "2.1.3")
+    args$steps <- as.integer(steps)
```
Comment (jjallaire) [inline R/model.R]: This needs to be `as_nullable_integer()` to account for a `NULL` values for `steps`

File: R/model.R [hunk 2]
```diff
@@ -525,28 +524,41 @@ predict.keras.engine.training.Model <- function(object, x, batch_size=NULL, verb
...
-  object$predict_classes(
+predict_classes <- function(object, x, batch_size = NULL, verbose = 0, steps = NULL) {
+  args <- list(
     x = keras_array(x),
     batch_size = as.integer(batch_size),
     verbose = as.integer(verbose)
   )
+
+  if (keras_version() >= "2.1.3")
+    args$steps <- as.integer(steps)
```
Comment (jjallaire) [inline R/model.R]: Same here (`as_nullable_integer()`)

**Comment by jjallaire** [comment]
There are also some tests failing due to code/documentation mismatches.

**Comment by jjallaire** [comment]
It looks like `batch_size = NULL` is causing a test to fail on Travis. Note that when running on Keras 2.1.4 locally I also see the same test failure:

```
✖ | 10 1     | model [3.4 s]
──────────────────────────────────────────────────────────
???: failure: models can be fit, evaluated, and used for predictions
`force(expr)` threw an error.
Message: TypeError: unsupported operand type(s) for +: 'int' and 'list'

Detailed traceback: 
  File "/Users/jjallaire/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/models.py", line 1112, in predict_proba
    preds = self.predict(x, batch_size, verbose, steps=steps)
  File "/Users/jjallaire/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/models.py", line 1025, in predict
    steps=steps)
  File "/Users/jjallaire/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/engine/training.py", line 1842, in predict
    verbose=verbose, steps=steps)
  File "/Users/jjallaire/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/engine/training.py", line 1325, in _predict_loop
    batches = _make_batches(num_samples, batch_size)
  File "/Users/jjallaire/.virtualenvs/r-tensorflow/lib/python2.7/site-packages/keras/engine/training.py", line 353, in _make_batches
    num_batches = (size + batch_size - 1) // batch_size  # round up
```

**Comment by jjallaire** [comment]
The issue may be that you needed `as_nullable_integer` here as well: https://github.com/rstudio/keras/blob/bugfix/keras-2.1.3-params/R/model.R#L539

**Comment by javierluraschi** [comment]
taking a look at the TF failures...

**Comment by jjallaire** [comment]
This fails using the tensorflow implementation because it uses an older version of keras. What we need to do here to preserve compatibility with all versions of keras is explicitly set to `k_epsilon()` only when  Keras < 2.1.3 (or whatever version this was introduced in). i.e. we will see this same failure in earlier versions of Keras that aren't the tensorflow implementation unless we do this.

**Pull request merged**